hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -382,6 +401,214 @@ deriving instance (IsTx tx, IsChainState tx) => FromJSON (Outcome tx)
 instance (IsTx tx, Arbitrary (ChainStateType tx)) => Arbitrary (Outcome tx) where
   arbitrary = genericArbitrary
 
+data HeadStateEvent tx
```

Comment:
Could we use the events returned by the update function to _create_ the effects and stop returning them?